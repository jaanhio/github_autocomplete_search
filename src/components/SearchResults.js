import React from 'react';
import styled from 'styled-components';

const SearchResultsGrid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
  width: 90vw;
  margin-top: 50px;
  column-gap: 1em;
`;

const ResultTitleLink = styled.a`
  color: #2769CF;
  overflow-wrap: break-word;
`;

const ResultFullName = styled.p`
  font-size: 0.8em;
  font-weight: 700;
  overflow-wrap: break-word;
`;

const ResultDescription = styled.p`
  font-size: 0.8em;
  overflow-wrap: break-word;
`;

const SearchResults = ({ searchResults }) => {
  const renderSearchResults = searchResults.length > 0 ? searchResults.map(result => {
    return (
      <div key={result.node_id}>
        <ResultTitleLink href={result.html_url} target='_blank'>{result.name}</ResultTitleLink>
        <ResultFullName>{result.full_name}</ResultFullName>
        <ResultDescription>{result.description}</ResultDescription>
      </div>
    )
  }) : null;
  return (
    <SearchResultsGrid>
      {renderSearchResults}
    </SearchResultsGrid>
  );
}

export default SearchResults;