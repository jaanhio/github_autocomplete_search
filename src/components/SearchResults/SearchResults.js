import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const SearchResultsGrid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
  width: 90vw;
  margin-top: 50px;
  column-gap: 1em;
`;

const ResultTitleLink = styled.a`
  color: #2769CF;
  overflow-wrap: break-word;
`;

const ResultFullName = styled.p`
  font-size: 0.8em;
  font-weight: 700;
  overflow-wrap: break-word;
  margin: 2px 0 0 0;
`;

const ResultDescription = styled.p`
  font-size: 0.8em;
  overflow-wrap: break-word;
  margin-top: 2px;
`;

const SearchResults = ({ searchResults, searchCategory }) => {
  const renderSearchResults = searchResults.length > 0 ? searchResults.map(result => {
    return searchCategory === 'repo' ? (
      <div key={result.node_id}>
        <ResultTitleLink href={result.html_url} target='_blank'>{result.name}</ResultTitleLink>
        <ResultFullName>{result.full_name}</ResultFullName>
        <ResultDescription>{result.description}</ResultDescription>
      </div>
    ) : (
        <div key={result.node_id}>
          <ResultTitleLink href={result.html_url} target='_blank'>{result.login}</ResultTitleLink>
        </div>
      )
  }) : null;
  return (
    <SearchResultsGrid>
      {renderSearchResults}
    </SearchResultsGrid>
  );
}

SearchResults.propTypes = {
  searchResults: PropTypes.array.isRequired,
  searchCategory: PropTypes.string.isRequired
};

SearchResults.defaultProps = {
  searchResults: [],
  searchCategory: 'repo'
}

export default SearchResults;