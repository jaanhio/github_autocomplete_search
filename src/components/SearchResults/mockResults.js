const mockRepoResults = [
  {
    "node_id": "12345678",
    "html_url": "www.google.com",
    "name": "result1",
    "full_name": "full_result1",
    "description": "result1_description"
  },
  {
    "node_id": "87654321",
    "html_url": "www.google.com",
    "name": "result2",
    "full_name": "full_result2",
    "description": "result2_description"
  },
];

const mockUserResults = [
  {
    "node_id": "12345678",
    "html_url": "www.google.com",
    "login": "user1",
  },
  {
    "node_id": "87654321",
    "html_url": "www.google.com",
    "login": "user2",
  }
]

module.exports = {
  mockRepoResults,
  mockUserResults
}