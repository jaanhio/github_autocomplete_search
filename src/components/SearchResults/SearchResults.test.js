import React from 'react';
import SearchResults from './SearchResults';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { mockRepoResults, mockUserResults } from './mockResults';

describe('<SearchResults /> component', () => {
  it('render without throwing error', () => {
    const wrapper = shallow(<SearchResults />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  it('renders null with empty searchResults array props', () => {
    const wrapper = shallow(<SearchResults />);
    expect(wrapper.props().children).toEqual(null);
  });
  it('renders repo searchResults correctly', () => {
    const wrapper = shallow(<SearchResults searchResults={mockRepoResults} />);
    const mockResultComponent = wrapper.props().children[0].props;
    expect(mockResultComponent.children.length).toEqual(3);
  });
  it('renders user searchResults correctly', () => {
    const wrapper = shallow(<SearchResults searchResults={mockUserResults} searchCategory='user' />);
    const mockResultComponent = wrapper.props().children[0].props;
    expect(mockResultComponent.children).toBeDefined();
  });
});