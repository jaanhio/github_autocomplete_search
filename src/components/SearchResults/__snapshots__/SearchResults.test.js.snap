// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> component render without throwing error 1`] = `<styled.div />`;
